simulator/options.mk

```diff
@@ -1,9 +1,13 @@
 # C++ compile
 CXX ?= g++
-CXXFLAGS:= -Wall -Wextra -Werror -Wpedantic -Wold-style-cast -Weffc++ -std=c++17
 CXXVERSION:= $(shell $(CXX) --version | grep version | sed -e 's/.*version //' -e 's/ .*//')
 
 UNAME:= $(shell uname -o)
+ifeq ($(UNAME), Darwin)
+        CXXFLAGS:= -Wall -Wextra -Werror -Wpedantic -Wold-style-cast -Weffc++ -std=c++1z
```

Comment:
This is a very bad fix.

Reply:
Well, since i don't have an osx machine, i have absolutely no idea which compiler and standard versions it supports. The latest information i managed to find on the internet was dated 23 Aug 2016 and said that osx clang doesn't support c++17, but some features will work with c++1z. If that really is the case i see only three options: reverting to c++1z, reverting to boost::string_ref, or reverting to a standalone implementation of string_view.

Reply:
Ok. There should be a way to update compiler with `homebrew` — I'll try to figure out with project alumni who uses Mac.
Can you please move CXXFLAGS to its original place and append with `-std=c**` depending on shell uname? I'd like to see some comment in `options.mk` why this is needed.